src/ln/channel.rs

```diff
@@ -106,19 +106,19 @@ enum OutboundHTLCState {
 	Committed,
 	/// Remote removed this (outbound) HTLC. We're waiting on their commitment_signed to finalize
 	/// the change (though they'll need to revoke before we fail the payment).
-	RemoteRemoved,
+	RemoteRemoved(Option<HTLCFailReason>),
 	/// Remote removed this and sent a commitment_signed (implying we've revoke_and_ack'ed it), but
 	/// the remote side hasn't yet revoked their previous state, which we need them to do before we
 	/// can do any backwards failing. Implies AwaitingRemoteRevoke.
 	/// We also have not yet removed this HTLC in a commitment_signed message, and are waiting on a
 	/// remote revoke_and_ack on a previous state before we can do so.
-	AwaitingRemoteRevokeToRemove,
+	AwaitingRemoteRevokeToRemove(Option<HTLCFailReason>),
 	/// Remote removed this and sent a commitment_signed (implying we've revoke_and_ack'ed it), but
 	/// the remote side hasn't yet revoked their previous state, which we need them to do before we
 	/// can do any backwards failing. Implies AwaitingRemoteRevoke.
 	/// We have removed this HTLC in our latest commitment_signed and are now just waiting on a
 	/// revoke_and_ack to drop completely.
-	AwaitingRemovedRemoteRevoke,
+	AwaitingRemovedRemoteRevoke(Option<HTLCFailReason>),
```

Comment:
Given that a state flag misuse has been and could be a likely source of errors, maybe we should have a more verbose name to dissociate AwaitingRemoteRevokeToRemote and AwaitingRemovedRemoteRevoke, like AwaitingRemoteRevokeToFinalDrop ? Had to read again their definition to get the slightly difference between them.

Reply:
Sure! That's somewhat orthogonal, though, so I'll leave that for a later cleanup, see #323.